### PR TITLE
Instagram Basic Display (Independent Publisher) : Add Refresh token endpoint to allow refresh of long lived token from instagram basic display

### PR DESCRIPTION
**Pull request by reshmee011**
---
- [ ] I attest that the connector doesn't exist on the Power Platform today. I've verified by checking the pull requests in GitHub and by searching for the connector on the platform or in the documentation. 
- [ ] I attest that the connector works and I verified by deploying and testing all the operations. 
- [ ] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [ ] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [ ] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [ ] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process:
- [ ] I have named this PR after the pattern of "Connector Name (Independent Publisher)" ex: HubSpot Marketing (Independent Publisher)
- [ ] Within this PR markdown file, I have pasted screenshots that show: 3 unique operations (actions/triggers) working within a Flow. This can be in one flow or part of multiple flows. For each one of those flows, I have pasted in screenshots of the Flow succeeding. 
- [ ] Within this PR markdown file, I have pasted in a screenshot from the Test operations section within the Custom Connector UI.
- [ ] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md. 

image

Added the refresh token endpoint to the connector to allow end users to refresh long lived token.

### CODE REVIEW

File: independent-publisher-connectors/InstagramBasicDisplay/apiDefinition.swagger.json [hunk 1]
```diff
@@ -266,6 +266,58 @@
...
+                                "type": "integer",
+                                "description": "The number of seconds until the long-lived token expires.",
+                                "title": "Expires In",
+                                "format": "int32"
+                            }
+                        }
+                    }
+                }
+            },
+            "summary": "Refreshes unexpired long-lived Instagram User Access tokens.",
```
Comment (vmanoharas) [inline independent-publisher-connectors/InstagramBasicDisplay/apiDefinition.swagger.json]: Summary should not end with period (.).  Please remove the period.

File: independent-publisher-connectors/InstagramBasicDisplay/apiDefinition.swagger.json [hunk 2]
```diff
@@ -266,6 +266,58 @@
           }
         ]
       }
+    },
+    "/refresh_access_token": {
+        "get": {
+            "responses": {
+                "default": {
```
Comment (vmanoharas) [inline independent-publisher-connectors/InstagramBasicDisplay/apiDefinition.swagger.json]: Please check below swagger validation error and fix the same. 

The 'default' response should not have schema definition. Schemas should be defined on expected responses only. ValuePath: /paths/~1refresh_access_token/get/responses/default

File: independent-publisher-connectors/InstagramBasicDisplay/apiDefinition.swagger.json [hunk 3]
```diff
@@ -266,6 +266,58 @@
...
+                                "description": "The number of seconds until the long-lived token expires.",
+                                "title": "Expires In",
+                                "format": "int32"
+                            }
+                        }
+                    }
+                }
+            },
+            "summary": "Refreshes unexpired long-lived Instagram User Access tokens.",
+            "description": "Long-lived tokens are valid for 60 days and can be refreshed as long as they are at least 24 hours old but have not expired, and the app user has granted your app the instagram_graph_user_profile permission. Refreshed tokens are valid for 60 days from the date at which they are refreshed. Tokens that have not been refreshed in 60 days will expire and can no longer be refreshed.",
```
Comment (vmanoharas) [inline independent-publisher-connectors/InstagramBasicDisplay/apiDefinition.swagger.json]: would suggest you keep all parameters descriptions short to fit in the text box. please see [here](https://learn.microsoft.com/en-us/connectors/custom-connectors/connector-string-guidance#parameters).

**Review by vmanoharas — COMMENTED** [review]
Hi @reshmee011, 

Please address review comments

**Comment by reshmee011** [comment]
@vmanoharas: Thanks for the review. I think I have addressed the review comments. Please check and let me know.

File: independent-publisher-connectors/InstagramBasicDisplay/apiDefinition.swagger.json (hunk 1 above)
Comment (vmanoharas) [reply independent-publisher-connectors/InstagramBasicDisplay/apiDefinition.swagger.json]: The dot (.) is still present in the summary, please remove, swagger validation is still failing with this error. Additionally please fix this error all over the swagger file.

File: independent-publisher-connectors/InstagramBasicDisplay/apiDefinition.swagger.json (hunk 2 above)
Comment (vmanoharas) [reply independent-publisher-connectors/InstagramBasicDisplay/apiDefinition.swagger.json]: please remove the value default for the description.

**Review by vmanoharas — COMMENTED** [review]
Hello @reshmee011, 

Previous issues are still exist, kindly fix them as per the review comment.

File: independent-publisher-connectors/InstagramBasicDisplay/apiDefinition.swagger.json (hunk 1 above)
Comment (reshmee011) [reply independent-publisher-connectors/InstagramBasicDisplay/apiDefinition.swagger.json]: The summary does not end with . anymore and I can't find any other similar errors.
image

File: independent-publisher-connectors/InstagramBasicDisplay/apiDefinition.swagger.json (hunk 2 above)
Comment (reshmee011) [reply independent-publisher-connectors/InstagramBasicDisplay/apiDefinition.swagger.json]: I have removed the default value.

**Comment by reshmee011** [comment]
Hello @vmanoharas , I am not sure why you can't see my changes, is it worth for me to close the PR and reopen another one.

**Review by vmanoharas — COMMENTED** [review]
Hello @reshmee011, 

Kindly check the below validation error messages and fix the same. 

##[error]The 'summary' value must end with alphanumeric character. ValuePath: /paths/~1refresh_access_token/get/summary

##[error]The 'default' response should not have schema definition. Schemas should be defined on expected responses only. ValuePath: /paths/~1refresh_access_token/get/responses/default

**Comment by reshmee011** [comment]
I am not sure despite I have made the required changes to resolve the errors, it's not visible/accessible on your end. I will close the PR to open another one.

**Pull request closed without merging**